Update Docs For Bound Audience Explanation (#30519)
hashicorp/vault: website/content/docs/auth/jwt/index.mdx

@include 'x509-sha1-deprecation.mdx'

~> **Note**: Starting in Vault 1.17, if the JWT in the authentication request

> contains an `aud` claim (more often the case than not), the associated `bound_audiences` for the "jwt" role
> must **exactly** match at least one of the `aud` claims declared for the JWT. For

Reviewer: ```suggestion
contains an `aud` claim (typical case) the associated `bound_audiences` for the
"jwt" role must **exactly** match at least one of the `aud` claims declared for
the JWT. For
```
